### PR TITLE
feat(secrets): Omit secrets from reports based on secrets reports

### PR DESCRIPTION
**Pull request by Eliran-Turgeman**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

SecretsOmitter will work for users using an API key & enabled secrets runner.

This module will process all reports and clear secrets from the code block of non-secrets records.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by arielkru — APPROVED** [review]
Very nice 💯

**Review by nimrodkor — APPROVED** [review]
Fix lint please

File: checkov/common/util/secrets_omitter.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        if not self.reports or not self.secrets_report:
+            logging.debug("Insufficient reports to omit secrets")
+            return SecretsOmitterStatus.InsufficientReports
+
+        files_with_secrets: set[str] = {secret_check.file_path for secret_check in self._secret_check()}
+        for check in self._non_secret_check():
+            check_file_path = check.file_path
+            check_line_range = check.file_line_range
+
+            if check_file_path not in files_with_secrets:
```
Comment (marynaKK) [inline checkov/common/util/secrets_omitter.py:67]: 🥇

File: checkov/common/util/secrets_omitter.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            # Should not reach here, used for typing
+            return
+
+        for check in self.secrets_report.failed_checks:
+            yield check
+
+    def _non_secret_check(self) -> Iterator[Record]:
+        for report in self.reports:
+            checks = report.failed_checks + report.passed_checks
+            for check in checks:
```
Comment (gruebel) [inline checkov/common/util/secrets_omitter.py:35]:
```suggestion
            for check in itertools.chain(report.failed_checks, report.passed_checks):
```
no need to create a new list

**Pull request merged**